set currLabels when currEntry is set.

woodsaj opened the pull request:
fixes #307

tomwilkie commented on pkg/iter/iterator.go:
```diff
@@ -166,7 +166,9 @@ func (i *heapIterator) Next() bool {
-	i.currEntry = mostCommon(tuples).Entry
+	curr := mostCommon(tuples)
+	i.currEntry = curr.Entry
+	i.currLabels = curr.EntryIterator.Labels()
```
This can be done in `NewHeapIterator`, as there is one heap iterator per stream and labels are constant in a stream.

woodsaj commented:
@tomwilkie i have updated the PR to set the labels in NewHeapIterator and updated the unit tests to ensure the labels are being returned correctly.

tomwilkie commented:
LGTM!
